src/network/mcpe/NetworkSession.php

```diff
@@ -1163,6 +1169,25 @@ public function onOpenSignEditor(Vector3 $signPosition, bool $frontSide) : void{
 		$this->sendDataPacket(OpenSignPacket::create(BlockPosition::fromVector3($signPosition), $frontSide));
 	}
 
+	/**
+	 * @param EnchantmentOption[] $options
+	 */
+	public function sendEnchantOptions(array $options) : void{
+		$protocolOptions = [];
+		$optionId = 0;
+
+		foreach($options as $option){
+			$protocolEnchantments = array_map(
+				fn(EnchantmentInstance $e) => new Enchant(EnchantmentIdMap::getInstance()->toId($e->getType()), $e->getLevel()),
+				$option->getEnchantments()
+			);
+			$protocolOptions[] = new EnchantOption($option->getRequiredXpLevel(), $optionId, $protocolEnchantments, [], [], $option->getDisplayName(), $optionId);
```

Comment:
The option IDs used here are problematic. In vanilla BDS, the IDs used are higher than all of the existing crafting recipes.

Some problems caused by this:
- In ItemStackRequestExecutor, we may accidentally select a crafting recipe instead of an enchant option if the server changed the player's current window before the transaction arrived. Instead, the ID should cause no recipe to be selected.
- Potentially, this might cause crafting recipes to get overwritten on the client side, potentially breaking crafting of some items. I haven't verified that this is the case yet, but I don't think this is a good idea to risk.

One possible solution is to offset the recipe IDs with some large value (e.g. 100_000) to prevent them from conflicting with normal recipes. However, this might break if lots of new recipes are introduced in the future.